apps/server/src/modules/provisioning/strategy/tsp/tsp.strategy.ts

```diff
@@ -108,7 +108,7 @@ export class TspProvisioningStrategy extends ProvisioningStrategy {
 				.filter(Boolean) as RoleName[],
 		});
 
-		if (externalUserDto.roles && externalUserDto.roles.length < 1) {
+		if (Array.isArray(externalUserDto.roles) && externalUserDto.roles.length < 1) {
```

Comment:
This still feels weird to me. roles is a non-nullable field and always an array of RoleNames, thus it should always exist and be an array.